libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -129,23 +130,22 @@ internal class MapboxRerouteController(
                     }
                 }
 
-                override fun onRoutesRequestFailure(
-                    throwable: Throwable,
+                override fun onFailure(
+                    reasons: List<RouterFailure>,
                     routeOptions: RouteOptions
                 ) {
                     logger.e(
```

Comment:
In any case, we're already logging in `MapboxDirectionsSession`, do we want to log again here?

Same for `onCanceled` 👇

Reply:
I think this one should stay. Developers can't hook into a failure of our default reroute controller, so we need to have a way to communicate the errors.

Reply:
Ah this is not true, developers can use `mapboxNavigation.getRerouteController()?.registerRerouteStateObserver` :+1: